Draft security proposal numfocus (#7)
jupyter/security: nf-security-committee.md

# Security Subcommittee for Numfocus

> While NumFOCUS and all it's affiliated and sponsored project are focused on

Reviewer: Looks like an incomplete sentence, should you just drop the "while" here and adjust?

> While NumFOCUS and all it's affiliated and sponsored project are focused on

Reviewer 2: ```suggestion
Typically, NumFOCUS and all its affiliated and sponsored projects are focused on
```

community and strive to communicate in public.

Security in contrast often requires private communication, and timely responses,
which is at odds with the typical open-source communication channels. Setting up
specific communication channels for each subproject can be costly for projects
in term of time invested for documentation and monitoring, especially for
subprojects. 

The multiple projects and different ways of inquiring about security can make it
hard for stakeholder to find how to contact the relevant projects; especially
when a question or vulnerability is related to multiple project interacting
together. 

> 

Reviewer: Some of these are maybe restating/expanding on the points listed:

- Discussions about best practices for security in general and at a project level are good to have as a community and in public.  People need to be educated and informed about security in a general sense and should be able to find resources they need deploy and run software securely.
- At the same time, developers need to coordinate activity in anticipation of and in response to software issues to do with security, in a time-sensitive fashion, and with discretion (it is just the reality).  Each project can do this on its own but the structures needed are basically the same from project to project.

In addition, I thought (restating but also expanding a bit maybe):

- Sharing structures (both infrastructure and human resources) for managing security across multiple projects can make it more manageable but also enable better propagation and awareness of issues across a set of projects that are often linked together or may encounter similar issues.
- A common structure may also promote a coherent approach to certain common security issues across projects, and forming strategies around those common issues actually can make running and deploying software securely easier ("these are NumFocus projects and they tend to do things this way...").

We ask NumFOCUS board for the creation of a security subcommittee. 

- This subcommittee should have at least 5 members selected by the NF board. 
- This subcommittee should be reachable via security@numfocus.org. 

The following is expected from the member of this subgroup:

- to acknowledge security questions, concern or vulnerability in a short
  timeframe (maximum 1 week), and contact the relevant(s) subprojects, or
  subproject(s) maintainers.

- for a security vulnerability report: ensuring the reporter get a proper
  response. The response can be directly from the concerned subproject. The 
  security subcommittee can directly provide a response if:

    - The subcommittee members have sufficient expertise to provide a response 
    - The subprojects maintainers do not reply to inquiries – in which the
      matter should also be discussed with numfocus board.

- Ignore any non-security enquiries, warn and ban repeated offender from
  contacting security@numfocus.org as their discretion.

> - Maintain a https://nufocus.org/security.txt text document following standard

Reviewer: typo, nufocus -> numfocus

security committee, work group, or contact channels for security related matter.

This subcommittee is not expected to directly work on improving security, or
defining security best practices.

We suggest the following 5 members for the initial subcommittee:

 - Volunteer 1 (affiliation, subproject if relevant), <email1@mail.com>
 - Volunteer 2 (affiliation, subproject if relevant), <email2@mail.com>
 - Volunteer 3 (affiliation, subproject if relevant), <email3@mail.com>
 - Volunteer 4 (affiliation, subproject if relevant), <email4@mail.com>
 - Volunteer 5 (affiliation, subproject if relevant), <email5@mail.com>

# Current project security process:

Many existing projects use tide lift, though this requires project maintainers
to be subscribed to tidelift – I'm not certain of all the implication.

- Numpy 
- PyMC3 
- Pandas
- SciPy
- Bokeh
- Dask

A few project s use custom mailing lists

- Julia : security@julialang.org.
- Matplotlib https://github.com/matplotlib/matplotlib/blob/master/SECURITY.md
  matplotlib@numfocus.org
- Jupyter / IPython (security@ipython.org)

For the following I was not able to find how to contact.

> - Nteract ??

Reviewer 2: ```suggestion
- Nteract https://github.com/nteract/nteract/security/policy
```
